cloudpub/ms_azure/service.py

```diff
@@ -241,7 +243,21 @@ def list_products(self) -> List[ProductSummary]:
             self._products = [p for p in self.products]
         return self._products
 
-    def get_product(self, product_id: str, first_target: str = "preview") -> Product:
+    def get_productid(self, product_name: str) -> str:
+        """Retrieve the desired product ID for the requested product name.
+
+        Args:
+            product_name (str): the product's name to retrieve its product ID.
+        Returns:
+            The requested product ID when found.
+        Raises NotFoundError when the product was not found.
+        """
+        for product in self.list_products():
```

Comment:
There is caching in the slef.list_products
* https://github.com/release-engineering/cloudpub/blob/0fa3a5ca3c6b2ad729ce05d0aae9709a33bc30a9/cloudpub/ms_azure/service.py#L242-243

I am not sure whether it is good idea to cache such data.
Is there any chance it could cause issues when we try to gracefully handle updates to the same offer from different processes (pub tasks)?